### ISSUE TITLE
[BUG] [Sentry middleware] 'ctx' circularly references itself in mapped type

### ISSUE THREAD

**Issue by Ninoks** [inngest/inngest-js#719]
**Describe the bug**
I tried to setup the sentry middleware as shown in the docs, but I got this typescript error :
```
src/background-functions/inngest/inngestClient.ts:39:30 - error TS2615: Type of property 'ctx' circularly references itself in mapped type '{ readonly steps: Readonly<{ id: string; data?: any; error?: any; }>[]; readonly fn: Any; readonly reqArgs: readonly unknown[]; ctx: any; }'.

 39 export const inngestClient = new Inngest({
                                 ~~~~~~~~~~~~~
 40   id: 'h',
    ~~~~~~~~~~~~~~~~~~~~
... 
 42   middleware: [sentryMiddleware()],
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 43 })
    ~~
```

**To Reproduce**
Init the inngestClient like so :
```
import { sentryMiddleware } from '@inngest/middleware-sentry'
import { EventSchemas, Inngest } from 'inngest'

export const inngestClient = new Inngest({
  id: 'h',
  schemas: new EventSchemas().fromRecord<Events>(),
  middleware: [sentryMiddleware()],
})
```

Init sentry before like so :
```
class App {
  public app: express.Application
  public env: string
  public port: string | number

  constructor(routes: Routes[]) {
    this.app = express()
    Sentry.init({
      dsn: process.env.SENTRY_URL,
      integrations: [Sentry.prismaIntegration()],

      tracesSampleRate: 0.3,
      profilesSampleRate: 0.3,
    })
    this.env = NODE_ENV || 'development'
    this.port = PORT || 3000

    this.initializeMiddlewares()
    this.initializeRoutes(routes)
    this.initializeErrorHandling()
  }
  ```

Using `"@sentry/node": "^8.33.1",` and  `"@inngest/middleware-sentry": "^0.1.1",`

**Expected behavior**
It should not throw any typescript error?

**Comment by jpwilliams** [comment]
Hi @Ninoks! Thanks for the report.

Can I ask what version of TypeScript you're using? This can sometimes be an issue with earlier versions.

**Comment by Ninoks** [comment]
Hi @jpwilliams, it's `4.5.2` currently.

**Comment by jpwilliams** [comment]
@Ninoks Ah, that might be the problem! We currently only support TS >=4.7, as many parts of the SDK use typing that's only supported in these versions.

Upgrading to `~4.7.0` (or to the latest `~5.6.0`) should resolve your issue!

**Comment by Ninoks** [comment]
Hey, thanks! 

Fixed using version `5.6.3`.
fyi the `4.9.5` had the same error.

**Comment by jpwilliams** [comment]
Ah - thanks for the heads up. Linking #637 - it seems to affect some particular circumstances. I may update the `@inngest/middleware-sentry` package to require TS 5.1.
